csrc/scheduler/matmul.cpp

```diff
@@ -696,9 +696,17 @@ void scheduleFusionInputsForEpilogue(
 } // namespace
 
 void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
-  static const bool should_unroll = true;
-  auto cached_and_forked_outputs =
-      scheduler_utils::cacheAndForkOutputs(fusion, should_unroll);
+  FusionGuard fg(fusion);
+
+  // Make sure we don't have global memory set on intermediate tensors from
+  // fusion segmentation
+  scheduler_utils::clearMemorySpace(fusion);
+
+  // Cache inputs
+  auto cached_inputs = scheduler_utils::cacheInputs(fusion, true);
```

Comment:
We don't use the `cached_inputs` variable
```suggestion
  scheduler_utils::cacheInputs(fusion, true);
```

Reply:
fixed